### PR TITLE
fix text obfuscation logic

### PR DESCRIPTION
**Pull request by Vadman97**
## Summary

innerHTML changes would not be obfuscated properly in default obfuscation mode.
https://github.com/highlight/rrweb/pull/116

fixes HIG-3970

## How did you test this change?

before
Screenshot from 2024-02-13 09-49-45

after
Screenshot from 2024-02-13 09-54-40

## Are there any deployment considerations?

new changeset

## Does this work require review from our design team?

no

### CODE REVIEW

File: frontend/src/pages/Buttons/Buttons.tsx [hunk 1]
```diff
@@ -129,6 +130,48 @@ export const Buttons = () => {
...
+								.toString(36)
+								.substring(7)}`,
+						)
+					}}
+				>
+					set email
+				</button>
+				<div>
+					<b>hello, this is a b tag</b>
+					<b key="email-1">{email}</b>
```
Comment (SpennyNDaJets) [inline frontend/src/pages/Buttons/Buttons.tsx:160]: @Vadman97 have you been able to reproduce the bug? I've been messing with some of the state, but the elements are being obfuscated through the `adds` nodes rather than the `texts`. Been messing with some css to see if I can reproduce better

**Review by SpennyNDaJets — APPROVED** [review]
Overall think this is the fix we want. Not sure if we want to include in this PR but from our conversation, we still need to:
- [ ] Confirm our edge case
  - incremental update through `texts`, only been able to update through `adds` which was working properly
  - see comment above
- [ ] Add end-to-end tests

File: frontend/src/pages/Buttons/Buttons.tsx (hunk 1 above)
Comment (Vadman97) [reply frontend/src/pages/Buttons/Buttons.tsx:160]: got it to repro not by using the react state, but by manually setting `innerText` see https://github.com/highlight/highlight/blob/669886629b7bfdefa079c5c097899a2e7ceabb6e/frontend/src/pages/Buttons/Buttons.tsx#L137-L141 and the screenshots above

Comment (Vadman97) [reply frontend/src/pages/Buttons/Buttons.tsx:160]: i'm guessing the customer's app is using jquery or some other library to directly edit the DOM innerText, hence we are seeing this bug for them

**Comment by Vadman97** [comment]
better e2e tests coming in https://linear.app/highlight/issue/HIG-4345/e2e-test-for-obfuscation
